app/scripts/views/settings/clients.js

```diff
@@ -115,41 +136,52 @@ define(function (require, exports, module) {
       });
     },
 
-    _onDeviceAdded: function () {
+    _onItemAdded: function () {
       this.render();
     },
 
-    _onDeviceRemoved: function (device) {
-      var id = device.get('id');
-      var self = this;
-      $('#' + id).slideUp(DEVICE_REMOVED_ANIMATION_MS, function () {
-        // re-render in case the last device is removed and the
-        // "no registered devices" message needs to be shown.
-        self.render();
-      });
+    _onItemRemoved: function (item) {
+      $('#' + item.get('id')).slideUp(DEVICE_REMOVED_ANIMATION_MS);
     },
 
-    _onDisconnectDevice: function (event) {
-      this.logViewEvent('disconnect');
-      var deviceId = $(event.currentTarget).attr('data-id');
-      this._destroyDevice(deviceId);
+    _onDisconnectClient: function (event) {
+      var itemId = $(event.currentTarget).data('id');
+      // type of client that was disconnected, can be 'client' or 'device'.
+      var clientType = $(event.currentTarget).data('type');
+
+      this.logViewEvent(clientType + '.disconnect');
+      if (clientType === 'device') {
+        this._destroyDevice(itemId);
+      } else if (clientType === 'app') {
+        this._destroyApp(itemId);
+      }
     },
 
-    _onRefreshDeviceList: function () {
+    _onRefreshClientsList: function () {
       var self = this;
       if (this.isPanelOpen()) {
         this.logViewEvent('refresh');
         // only refresh devices if panel is visible
         // if panel is hidden there is no point of fetching devices
-        this._fetchDevices().then(function () {
+        this._fetchAllClientTypes().then(function () {
```

Comment:
`ClientTypes` seems like the wrong noun here. I understand what you are trying to convey, it's that like `App`, `Client` has had a very specific meaning up until now. Would `ConnectedServices`, e.g., `fetchConnectedServices` be sufficiently different but specific enough? `ConnectedServicesAndDevices`?